Infusionsoft/InvoiceService.php

```diff
@@ -86,7 +91,7 @@ public static function chargeInvoiceArbitraryAmountUsingPayPlan($invoiceId, $car
             $daysBetweenPayments = 1;
         }
         try{
-            Infusionsoft_InvoiceService::addPaymentPlan($invoiceId, 0, $cardId, $merchantAccountId, 1, 1, $temporaryFirstPayment, Infusionsoft_App::formatDate(date('Y-m-d')), Infusionsoft_App::formatDate(date('Y-m-d', strtotime(' + ' . $daysBetweenPaymentsForChargePaymentPlan . ' days'))), $numberOfPaymentsForChargePaymentPlan, $daysBetweenPaymentsForChargePaymentPlan);
+            Infusionsoft_InvoiceService::addPaymentPlan($invoiceId, 0, $cardId, $merchantAccountId, 1, 1, $temporaryFirstPayment, Infusionsoft_App::formatDate($today), Infusionsoft_App::formatDate(date('Y-m-d', strtotime($today . ' + ' . $daysBetweenPaymentsForChargePaymentPlan . ' days'))), $numberOfPaymentsForChargePaymentPlan, $daysBetweenPaymentsForChargePaymentPlan);

```

Comment:
Also, I know you didn't do this, but can you change $temporaryFirstPayment to $temporaryFirstPaymentAmount